plugins/woocommerce-admin/client/products/sections/product-details-section.tsx

```diff
@@ -74,33 +81,34 @@ export const ProductDetailsSection: React.FC = () => {
 							{ ...getTextControlProps(
 								getInputProps( 'name' )
 							) }
+							onBlur={ () => {
+								setSkuIfEmpty();
```

Comment:
If you override the onBlur then you should call the parent onBlur manualy like:
```javascript
const nameTextControlProps = getTextControlProps( getInputProps( 'name' ) );
...
onBlur={ () => {.
    ...
    nameTextControlProps?.onBlur();
} }
```
Validations and other internal form states change when onBlur happened.

Reply:
Good catch, @mdperez86!  Updated in https://github.com/woocommerce/woocommerce/pull/34978/commits/b00dbf96c6e29e6cc3902db89a32bb5ff01fc2d2.

On a side note, I think the `getTextControl` and other wrappers around `getInputProps` are relatively thin at the moment.  I think it's also worth implementing some of this logic in the `Form` component to prevent all of these parent calls that we're going to see throughout this experience.

Reply:
Issue created to tackle the `getInputProps` changes made [here](https://github.com/woocommerce/woocommerce/issues/34996).

Reply:
> getTextControlProps( getInputProps( 'name' ) )

I don't like this. I prefer a Hoc or a [controller](https://react-hook-form.com/get-started/#IntegratingControlledInputs) component to adapt props. So the `getInputProps` or what ever the name should return all necesary props related to the form logic and not to the control we are registering to the form logic. 

> I think it's also worth implementing some of this logic in the Form component to prevent all of these parent calls that we're going to see throughout this experience.

Same page with you

Reply:
> Issue created to tackle the getInputProps changes made [here](https://github.com/woocommerce/woocommerce/issues/34996).

Awesome!!!!